processor_test.go

```diff
@@ -282,3 +313,307 @@ func (ts *testService) ProcessMsg(msg *testMsg) (network.Message, error) {
 	ts.Msg = msg
 	return msg, nil
 }
+
+func TestProcessor_REST_Registration(t *testing.T) {
+	h1 := NewLocalServer(tSuite, 2000)
+	defer h1.Close()
+	p := NewServiceProcessor(&Context{server: h1})
+	require.NoError(t, p.RegisterRESTHandler(procRestMsgGET1, "dummyService", "GET", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procRestMsgGET2, "dummyService", "GET", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procRestMsgGET3, "dummyService", "GET", 3, 3))
+
+	require.NoError(t, p.RegisterRESTHandler(procRestMsgPOSTString, "dummyService", "POST", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procRestMsgPOSTPoint, "dummyService", "POST", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procMsg, "dummyService", "POST", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procMsg2, "dummyService", "POST", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procMsg3, "dummyService", "POST", 3, 3))
+	require.NoError(t, p.RegisterRESTHandler(procMsg4, "dummyService", "POST", 3, 3))
+
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGET3, "dummyService", "GET", 3, 2))
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGET3, "dummyService", "GET", 1, 2))
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGETWrong1, "dummyService", "GET", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGETWrong2, "dummyService", "GET", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGETWrong3, "dummyService", "GET", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procRestMsgGET1, "dummyService", "XXX", 3, 3))
+
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong1, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong2, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong3, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong4, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong5, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong6, "dummyService", "POST", 3, 3))
+	require.Error(t, p.RegisterRESTHandler(procMsgWrong7, "dummyService", "POST", 3, 3))
+}
+
+type bnPoint struct {
+	P kyber.Point
+}
+
+type bnPointWrapper struct {
+	P []byte
+}
+
+func (p bnPoint) MarshalJSON() ([]byte, error) {
+	buf, err := p.P.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	return []byte(fmt.Sprintf(`{"P": "%s"}`, base64.StdEncoding.EncodeToString(buf))), nil
+}
+
+func (p *bnPoint) UnmarshalJSON(b []byte) error {
+	wrapper := bnPointWrapper{}
+	if err := json.Unmarshal(b, &wrapper); err != nil {
+		return err
+	}
+	suite := bn256.NewSuite()
+	g2 := suite.G2().Point()
+	if err := g2.UnmarshalBinary(wrapper.P); err != nil {
+		return err
+	}
+	p.P = g2
+	return nil
+}
+
+type edPoint struct {
+	P kyber.Point
+}
+
+type edPointWrapper struct {
+	P []byte
+}
+
+func (p edPoint) MarshalJSON() ([]byte, error) {
```

Comment:
So this is all very interesting, but I'm surprised to find it in *_test.go. Are you expecting to move it into Kyber later?

Reply:
`MarshalJSON` could be in kyber but having `UnmarshalJSON` in kyber is useless. suppose you want to unmarshal something, you cannot write `json.Unmarshal(buf, point)` where `point` is a `kyber.Point` because the unmarshaler doesn't know which implementation to use.